### PR TITLE
Use deepmerge for options

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

As we allow to having deep options structure, we forget to use deep merge to make sure, defaults are working as they should and not only on the root level.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #https://github.com/home-assistant/addons/issues/2202
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/addons/addon.py [hunk 1]
```diff
@@ -87,6 +88,12 @@
 
 WATCHDOG_TIMEOUT = aiohttp.ClientTimeout(total=10)
 
+_OPTIONS_MERGER: Final = Merger(
+    type_strategies=[(list, ["overwrite"]), (dict, ["merge"])],
```
Comment (ludeeus) [inline supervisor/addons/addon.py:92]:
```suggestion
    type_strategies=[(list, ["override"]), (dict, ["merge"])],
```

Comment (ludeeus) [reply supervisor/addons/addon.py:92]: This is bad, this will overwrite always.

File: supervisor/addons/addon.py [hunk 2]
```diff
@@ -87,6 +88,12 @@
...
+    fallback_strategies=["override"],
+    type_conflict_strategies=["override"],
+)
```
Comment (ludeeus) [inline supervisor/addons/addon.py:95]:
```suggestion
_OPTIONS_MERGER: Final = Merger(
    type_strategies=[(dict, ["merge"])],
    fallback_strategies=["use_existing"],
    type_conflict_strategies=["override"],
)
```

I _think_ this should work and be safe

File: tests/api/test_store.py [hunk 3]
```diff
@@ -14,8 +14,9 @@ async def test_api_store(
     resp = await api_client.get("/store")
     result = await resp.json()
 
-    assert result["data"]["addons"][0]["slug"] == store_addon.slug
-    assert result["data"]["repositories"][0]["slug"] == repository.slug
+    print(result)
```
Comment (ludeeus) [inline tests/api/test_store.py:17]:
```suggestion
```

**Review by ludeeus — APPROVED** [review]

**Review by frenck — APPROVED** [review]

**Pull request merged**